frontend-project/src/pages/institutions/new/index.jsx

```diff
@@ -0,0 +1,121 @@
+import { PageHeaderWrapper } from '@ant-design/pro-layout';
+import { Button, Col, Card, Form, Input, Row, Select } from 'antd';
+import React from 'react';
+import { formatMessage, FormattedMessage } from 'umi-plugin-react/locale';
+
+const layout = {
+  labelCol: { span: 8 },
+  wrapperCol: { span: 16 },
+};
+
+const tailLayout = {
+  wrapperCol: { offset: 8, span: 16 },
+};
+
+const InstitutionsNewForm = () => {
```

Comment:
Po zmienieniu pliku na .tsx zdefiniuj interfejs dla zwracanego elementu
https://www.typescriptlang.org/docs/handbook/jsx.html#react-integration

Reply:
Na razie komponent tylko renderuje widok, więc stworzyć pusty interfejs, czy już z przyszłymi typami? (aktualnie byłyby opcjonalne)

Reply:
Z tą konfiguracją typy i tak są opcjonalne, ich definiowanie jest za to dobrą praktyką.
W tym przypadku `InstitutionsNewForm` może mieć zdefiniowany typ za pomocą jednego z interfejsów dostarczanych przez react. W tym przypadku będzie to prawdopodobnie interfejs generyczny `Component` lub `FC` (Function Component).
Jak pisałem w odpowiedzi do @magul nie mam doświadczenia z `tsx`, jedynie z "czystym" TypeScriptem, więc nie jestem w stanie stwierdzić, który z interfejsów generycznych dostarczanych przez Reacta i inne biblioteki, które tutaj używamy w tym projekcie tutaj pasuje.

Jak masz jakieś pytania to pisz tutaj/na slacku Code For Poznań (@ Jakub Surdej)

Reply:
ok, dodałem FunctionalComponent do komponentu, interfejs z typami propsów, dodam na bieżąco wraz z ich pojawianiem się.